src/main/java/codeu/controller/ParsingEngine.java

```diff
@@ -0,0 +1,43 @@
+package codeu.controller;
+
+import codeu.model.data.BotRequest;
+import codeu.model.data.BotStructure;
+import codeu.model.data.Message;
+import codeu.model.data.RoutingEngine;
+
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+	
+
+public class parsingEngine {
```

Comment:
Style guide says UpperCamelCase for class names:
https://google.github.io/styleguide/javaguide.html#s5.2.2-class-names

That is: `ParsingEngine`